cpp/tests/io/json_writer.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
```

Comment:
I like that the tests are in the separate file from the reader tests. Other formats have reader and writer tests in the same file and it in 90% of cases it just leads to longer time to run all relevant tests.